### ISSUE TITLE
mpv.net Compatibility? 

### ISSUE THREAD

**Issue by asianmusicguy** [cyl0/MordenX#6]
@cyl0 
Every time I run MPV.net after installing the script and load the player this occurs
![image](https://user-images.githubusercontent.com/39034611/154341726-fdc71312-079f-4793-9ee4-20764ae05bd2.png)
 any idea how to make them play nice mpv and mpv.net are not that much different from one another

https://github.com/stax76/mpv.net

**Comment by asianmusicguy** [comment]
@asilentdreamer  so what can be done could @cyl0  fix it or?

**Comment by asianmusicguy** [comment]
Interesting if i block out logo it breaks the entire script from working @asilentdreamer  i have --function show_logo ()
	local osd_w, osd_h = 640, 360
	local logo_x, logo_y = osd_w/2, osd_h/2-20
	local ass = assdraw.ass_new--()

**Comment by asianmusicguy** [comment]
on top of this the osc wont disappear on pause if i add this script nor does mpv.nets toggle for it work :(  i rea;;y want this to work as mpv.nets default osc suck

**Comment by asilentdreamer** [comment]
If you remove the entire function the script will break because now there is a line in code that calls the function to draw the logo which now does not exist

The best option is to replace the function with this:
```lua
function show_logo()
	local osd_w, osd_h = 640, 360
	local logo_x, logo_y = osd_w/2, osd_h/2-20
	local ass = assdraw.ass_new()

	ass:new_event()
	ass:pos(logo_x, logo_y+70)
	ass:an(8)
	ass:append(texts.welcome)
	set_osd(osd_w, osd_h, ass.text)
end
```
This will also show the welcome text on the screen.
If you would rather have just the mpv.net logo and no welocme text you could remove the line that calls this function. For that just comment out the selected line in the screenshot below
![image](https://user-images.githubusercontent.com/69895746/154439907-7c93845f-8870-4849-aa26-42d9426fd25a.png)

**Comment by asianmusicguy** [comment]
@asilentdreamer  thank you all your help that did solve one issue but I still cant seem to toggle visibility via mpv net context menu  nor is a toggle message displayed and the controls are still visible on pause i want them off when paused. any ideas?

**Comment by asilentdreamer** [comment]
I dont really know the solution to those. Feel free to ping me if you find it.

**Comment by asianmusicguy** [comment]
@asilentdreamer I'm at a loss I gave opened a ticket with the original dev and the ticket here all i wanted was to replace the old OSC which is a little outdated   and modern was perfect you wouldn't think it be this hard right I have QA expertise and your a lua coder between us maybe we can figure something out I wouldn't expect you to spend a lot of time on it but maybe we can make both mpv net and modern a little better in the process who knows

**Comment by asilentdreamer** [comment]
I am not a lua coder. All I did is copy a few lines of code from mpv's osc and paste it in this osc. Also it looks like the blue mpv.net icon is hard coded so doesn't look like it can be replaced.

**Comment by asianmusicguy** [comment]
@asilentdreamer  that's fine because your solution worked now i just need to make the OSD disappear on pause as it does without modern installed and also display toggle status messages again for the corresponding toggle and the only one being the least bit helpful is you :) anyway its hard because i can find code to make the OSD stay because apparently it is meant ti do that natively but adding modern breaks the behavior for some reason.

**Comment by asilentdreamer** [comment]
That has been my experience too. I tried to add that but the script broke. So after a few tries I gave up because I understand neither the script nor the langauge very well.

**Comment by asianmusicguy** [comment]
@asilentdreamer  if you  know anyone who could possibly help that be awesome as neither developer seems interested in even guiding us and the documentation doesnt cover anything  like this

**Comment by dexeonify** [comment]
>  i just need to make the OSD disappear on pause

Open Config Folder, create a `script-opts` folder and make a new file call `osc.conf`. Add `showonpause=no` inside `osc.conf`.
Alternatively, you can just change the options directly in `mordenx.lua`, but that's not recommended.

> I still cant seem to toggle visibility via mpv net context menu

This was certainly fun to debug - the feature isn't implemented in MordenX :D. But don't worry, it's easy to add the feature, I should probably make a PR later.

Open `mordenx.lua`, go to line 2201 or search for `function visibility_mode(mode, no_osd)`.
Add this block of code inside the function, above `if mode == 'auto' then`:

```lua
function visibility_mode(mode, no_osd)
    if mode == "cycle" then
        if not state.enabled then
            mode = "auto"
        elseif user_opts.visibility ~= "always" then
            mode = "always"
        else
            mode = "never"
        end
    end

    if mode == 'auto' then
    ...
```

Now open `input.conf` inside the config folder, modify line 146
```
 Del     script-binding osc/visibility               #menu: View > Toggle OSC Visibility
```
to
```
 Del     script-binding mordenx/visibility           #menu: View > Toggle OSC Visibility
```
The reason is explained here: dexeonify/mpv-config@967413b

**Comment by asianmusicguy** [comment]
@dexeonify  I must have done something wrong because now t modern does not show up at all could you possibly upload the fixed files and ill  just put them in place or do we have to do more troubleshooting now

**Comment by dexeonify** [comment]
Hmm, I did encountered that a few times, but it was intermittent and it's not caused by my fix. Maybe try restarting mpv.net again? Any messages printed in the console?

**Comment by asianmusicguy** [comment]
@dexeonify  nothing that i can see its gone back to mpv.nets osc though

**Comment by asianmusicguy** [comment]
maybe i edited wrong but i dont think so

**Comment by dexeonify** [comment]
Yea, I'm not sure why it suddenly didn't work for you when it's working previously. Try reverting your changes and see which changes caused it.

Here's the mpv.net directory for reference. I didn't change anything major.
[mpv.net.zip](https://github.com/cyl0/MordenX/files/8104148/mpv.net.zip)

**Comment by asianmusicguy** [comment]
@dexeonify  Well im not sure now because it works flawless when i replaced the directors i must have just miss typed something the only thing i can think of is I did attempt to use @asilentdreamer 's fix to preserve on screen text where as yours does not maybe that plus your edits doesnt work but its all good now id still love to preserve text?

**Comment by asianmusicguy** [comment]
Lastly I noticed files don't automatically start on open now i actually have to hit play now i never used to ?

**Comment by dexeonify** [comment]
> I did attempt to use @asilentdreamer 's fix to preserve on screen text where as yours does not maybe that plus your edits doesnt work

Ah yea, I chose the 2nd option to completely hide the logo.

> Lastly I noticed files don't automatically start on open now i actually have to hit play now i never used to ?

That's because of the `pause = yes` and `mute = yes` I added in the `mpv.conf`. I only added those for easier debugging. Feel free to remove those. I did say my config is for reference. 😛

**Comment by asianmusicguy** [comment]
so all this this debugging will it benefit modernx in anyway as that was my ultimate goal to prevent others who happen to be using the mpv.net variation from going through this while still being able to enjoy a music better osc
@dexeonify

**Comment by dexeonify** [comment]
Yea, I guess. We still have to wait for the author to merge my PR (#7) so others don't have to dig into the code and add the features manually.

There's still 2 things others have to do manually for compatibility with mpv.net though:
- Remove the logo (see https://github.com/cyl0/MordenX/issues/6#issuecomment-1042712442)
- Toggle visibility via context menu (see https://github.com/cyl0/MordenX/issues/6#issuecomment-1046161224)

**Comment by asianmusicguy** [comment]
I think since between the author code do since between the 3 of us we have done all the leg work :( is just summarize what we have done and state if using the mpv varient  make these adjustments

**Comment by asianmusicguy** [comment]
either way thank you @asilentdreamer @dexeonify  we make a good team

**Comment by asianmusicguy** [comment]
@dexeonify  i have one more issue if you would not mind looking at when you get time?
with the new osc it is very hard to control the seek bar with the mouse select drag then click to release all seem very difficult

**Comment by asilentdreamer** [comment]
> either way thank you @asilentdreamer @dexeonify we make a good team

NP. The fixes mentioned in this issue were a lot of help

**Comment by dexeonify** [comment]
> it is very hard to control the seek bar with the mouse select drag then click to release all seem very difficult

Related issues stax76/mpv.net#350 stax76/mpv.net#195

**Comment by SonaliBendre** [comment]
tl;dr?

**Comment by dexeonify** [comment]
- Remove MordenX's logo (https://github.com/cyl0/MordenX/issues/6#issuecomment-1042712442)
- Fix seekbar dragging (remove `osc-scalewindowed=1.5` in `mpv.conf`)
- Make OSD disappear on pause (https://github.com/cyl0/MordenX/issues/6#issuecomment-1046161224)
- Toggle OSC visibility via <kbd>DEL</kbd> (https://github.com/cyl0/MordenX/issues/6#issuecomment-1046161224)

Only the first two points are mpv.net issues.

**Comment by SonaliBendre** [comment]
> * Remove MordenX's logo ([mpv.net Compatibility?  #6 (comment)](https://github.com/cyl0/MordenX/issues/6#issuecomment-1042712442))
> * Fix seekbar dragging (remove `osc-scalewindowed=1.5` in `mpv.conf`)
> * Make OSD disappear on pause ([mpv.net Compatibility?  #6 (comment)](https://github.com/cyl0/MordenX/issues/6#issuecomment-1046161224))
> * Toggle OSC visibility via DEL ([mpv.net Compatibility?  #6 (comment)](https://github.com/cyl0/MordenX/issues/6#issuecomment-1046161224))
> 
> Only the first two points are mpv.net-exclusive issues.

Please give me the modified complete working script and istruction what to change

**Comment by dexeonify** [comment]
...
The instructions are all there, but whatever.

Open mpv.net, right click > Settings > **Open Config Folder**. Download [MordenX-mpv.net-patched.zip](https://github.com/cyl0/MordenX/files/8115478/MordenX-mpv.net-patched.zip) and extract it inside the folder.

Open `input.conf` and modify line 146 to
```properties
 Del     script-binding mordenx/visibility           #menu: View > Toggle OSC Visibility
```

Then, open `mpv.conf` and modify line 9 to
```properties
script-opts = osc-hidetimeout=2000,console-scale=1
```

**Comment by SonaliBendre** [comment]
> ... The instructions are all there, but whatever.
> 
> Open mpv.net, right click > Settings > **Open Config Folder**. Download [MordenX-mpv.net-patched.zip](https://github.com/cyl0/MordenX/files/8115478/MordenX-mpv.net-patched.zip) and extract it inside the folder.
> 
> Open `input.conf` and modify line 146 to
> 
> ```ini
>  Del     script-binding mordenx/visibility           #menu: View > Toggle OSC Visibility
> ```
> 
> Then, open `mpv.conf` and modify line 9 to
> 
> ```ini
> script-opts = osc-hidetimeout=2000,console-scale=1
> ```

Thank you very much. This instruction is only for mpv.net right? (just confirming never mind)

**Comment by dexeonify** [comment]
Well, the instructions for `input.conf` can be applied to mpv as well.

**Comment by cyl0** [comment]
Hi there, sorry for the late response but I've merged @dexeonify's PR. I'm not able to test with mpv.net myself, so please test and see if the issue persists.

**Comment by cyl0** [comment]
I'll be closing this due to inactivity, feel free to add a comment here if this issue still persists.

**Comment by RibShark** [comment]
I am having another issue using this with mpv.net. Most of the time, the OSD does not respond to any mouse events (moving, clicking), and the mpv.net logo does not show at all. If I repeatedly open and close mpv.net until the logo shows properly, the mouse events will also work.
